extensions/amp-social-share/0.2/storybook/Basic.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {SocialShare} from '../social-share';
+import {addParamsToUrl} from '../../../../src/url';
+import {getSocialConfig} from '../amp-social-share-config';
+import {select, text, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+
+export default {
+  title: 'Social Share',
+  component: SocialShare,
+  decorators: [withA11y, withKnobs],
+};
+
+export const _default = () => {
+  const knobConfigurations = [
+    'email',
+    'facebook',
+    'linkedin',
+    'pinterest',
+    'tumblr',
+    'twitter',
+    'whatsapp',
+    'line',
+    'sms',
+    'system',
+    'custom endpoint',
+  ];
+  let type = select('Provider Type', knobConfigurations, knobConfigurations[0]);
+  let href = text('Custom Share Endpoint', 'Not Specified');
+
+  const config = getSocialConfig(type);
+  if (type !== 'custom endpoint') {
+    href = addParamsToUrl(config.shareEndpoint, config.defaultParams);
```

Comment:
Yes. I agree: the conflation of the endpoint and href is not entirely great here. I don't know all the dependencies there at this point, but this is way too much code for instrumenting the component.

Reply:
We should have a larger design discussion around modifications to this component to make sure we are all on the same page this time. As far as I understand there are several points to address here:

(1) There are two URLs we are dealing with, the one of the page that the user (and component) is on _to be shared_ and the one of the social media the user opens _to share on_. According to the component documentation on amp.dev: "It assumes that the current document canonical url is the URL you want to share and the page title is the text you want to share." These defaults can and should be moved to the Preact component.

(2) The preconfigured params take an arbitrary number of `data` attributes in AMP mode to construct the sharing endpoint. My understanding is we want the Preact component to construct the share endpoint, but AFAIK the restricting factor was with respect to passing the arbitrary `data` props from the AMP layer to the Preact layer. We need two utilities to make this happen: a `prop`-specific version of `getDataParamsFromAttributes` for the Component to read, and a utility for the AMP custom element to be able to pass its `data-` attributes as props to the component. 

(3) Some of the `data` params in (2) must also be able to override the defaults in (1). For example we need a way for the AMP custom element to override the default page URL retrieved in the Preact component with the canonical URL which is an AMP-specific concept.

Reply:
Right there is a lot of configuration needed for the preact version of the component.  Let's hold off on further changes to this story until we complete the re-design of the component.

Reply:
Yes. Let's proceed and we'll rework this as needed.